model/QtiJsonItemCompiler.php

```diff
@@ -122,9 +126,14 @@ protected function deployQtiItem(
             $data = $this->convertXmlAttributes($data);
             $this->itemJson['data'] = $data['core'];
             $metadata = $this->getMetadataProperties();
+            $enhancedMetadata = $this->getEnhancedMetadataProperties();
 
             $privateDirectory->write($language . DIRECTORY_SEPARATOR . self::ITEM_FILE_NAME, json_encode($this->itemJson));
             $privateDirectory->write($language . DIRECTORY_SEPARATOR . self::METADATA_FILE_NAME, json_encode($metadata));
+            $privateDirectory->write(
```

Comment:
Some questions: 

- Will we garbage collect this after? 
- We will always persist this data, regardless customer will use data-store or not? It will increate storage space.
- Should we have a kind of "flag" to allow safe this data or not?

Reply:
I think it's not only about persisting, but also retrieval all the associated metadata from storage,  can we do that only when it's really needed?